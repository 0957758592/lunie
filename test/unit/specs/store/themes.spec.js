import setup from "../../helpers/vuex-setup"

let instance = setup()

describe("Module: Themes", () => {
  let store, state

  beforeEach(() => {
    store = instance.shallow().store
    state = store.state.themes
  })

  it("loads themes", () => {
    expect(state.active).toBe("dark")
    localStorage.setItem("appTheme", "light")
    store.commit("loadTheme")
    expect(state.active).toBe("light")
  })

  it("sets themes", () => {
    expect(state.active).toBe("dark")
    store.commit("setTheme", "light")
    expect(state.active).toBe("light")
  })

  it("updates themes", () => {
    store.commit("updateTheme", "light")
  })
})
