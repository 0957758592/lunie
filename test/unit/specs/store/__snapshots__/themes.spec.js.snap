// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module: Themes has a dark theme 1`] = `
Object {
  "app-bg": "hsl(233, 37%, 18%)",
  "app-bg-alpha": "hsla(237, 18%, 20%, 95%)",
  "app-bg-light": "hsl(233, 40%, 14%)",
  "app-fg": "hsl(233, 43%, 10%)",
  "bc": "hsl(233, 22%, 23%)",
  "bc-dim": "hsl(233, 33%, 16%)",
  "bc-light": "#dfdfdf",
  "box-shadow": "hsla(0,0,0,0.2)",
  "dim": "hsl(240, 5%, 75%)",
  "hover-bg": "hsla(233, 88%, 90%, 0.2)",
  "input-bc": "hsl(233, 22%, 33%)",
  "input-bc-hover": "hsl(233, 22%, 40%)",
  "link": "hsl(233, 88%, 82%)",
  "txt": "hsl(0, 0%, 100%)",
}
`;

exports[`Module: Themes has a light theme 1`] = `
Object {
  "app-bg": "#f5f5f5",
  "app-bg-alpha": "hsla(233, 0%, 0%, 5%)",
  "app-bg-light": "#fff",
  "app-fg": "hsl(0,0%,92%)",
  "bc": "#ddd",
  "bc-dim": "#cdcbcb",
  "bc-light": "#dfdfdf",
  "box-shadow": "hsla(0,0,80%,0.2)",
  "dim": "#3b3b3b",
  "hover-bg": "hsla(233, 88%, 90%, 0.2)",
  "input-bc": "hsl(233, 22%, 67%)",
  "input-bc-hover": "hsl(233, 22%, 40%)",
  "link": "hsl(233, 88%, 57%)",
  "txt": "#000",
}
`;
